### ISSUE TITLE
Improve Jenkins builds to support `python nightly`

### ISSUE THREAD

**Issue by simitt** [elastic/apm-agent-python#20]
There is no `nightly` docker image available at [python docker images](https://hub.docker.com/_/python/). Investigate if there are other docker repositories offering python nightly builds, and add to Jenkins test handling if available.

**Comment by ruflin** [comment]
In case there are no images around, I could also see us building it our self. As this could take a bit of time I could see the tests which are run on the nightly build to be run as a follow up build step after things were merged into master to not run it on each PR commit.

**Comment by beniwohli** [comment]
I'd classify this as very low priority. We get it on Travis for free, and it's not super important to know it instantly when the nightly Python fails, so if Travis had an outage, we could still ship.

**Comment by kuisathaverat** [comment]
There is a `rc` docker image that I think it is what we are looking for

`3.8.0a3, 3.8-rc, rc`

**Comment by beniwohli** [comment]
@kuisathaverat that's not quite what I'd need. The alpha/RC docker images of the upcoming Python version are built relatively seldomly and late in the development process. Ideally, there would be a docker image that is built daily, with the newest master checkout from https://github.com/python/CPython. That way, we can detect regressions early.

But as long as we don't have a way to indicate allowed failures (parts of the build matrix that are allowed to fail without making the whole build fail), testing against nightly builds is not a good idea anyway IMO (a regression in a nightly build would force us to drop everything else to find a fix so we can make the build green again).

**Comment by kuisathaverat** [comment]
agree, test against a master branch could be a pain because of that I purpose the RC Docker image, it'd be something near to a production release so something stable, and give you some weeks to detects regresions.
